src/text-buffer.coffee

```diff
@@ -320,6 +323,22 @@ class TextBuffer
   getPath: ->
     @file?.getPath()
 
+  # Public: Set the associated file for the buffer.
+  #
+  # * `file` A {File} representing the file
+  setFile: (file) ->
+    return if file?.getPath() == @getPath()
```

Comment:
Why shouldn't you be able to set the file when the buffer's path matches the file's?